subsys/net/lib/download_client/src/download_client.c

```diff
@@ -620,6 +625,11 @@ void download_thread(void *client, void *a, void *b)
 
 			rc = request_send(dl);
 			if (rc) {
+				if (dl->fd == -1) {
```

Comment:
Just wondering, shouldn't the event be issued anyway?

Reply:
What do you mean?

Reply:
I mean, now in case of error code the loop brakes before `ECONNRESET` even is issued to the application (just below).

Reply:
If the application cancelled the download client, I do not see any reason for issuing ECONNRESET. The download client should just stop immediately. In case we issue ECONNRESET this code will be executed in fota_download.c. I do not think that this code should be executed.

`	case DOWNLOAD_CLIENT_EVT_ERROR: {
		/* In case of socket errors we can return 0 to retry/continue,
		 * or non-zero to stop
		 */
		if ((socket_retries_left) && ((event->error == -ENOTCONN) ||
					      (event->error == -ECONNRESET) ||
					      (event->error == -ETIMEDOUT))) {
			LOG_WRN("Download socket error. %d retries left...",
				socket_retries_left);
			socket_retries_left--;
			/* Fall through and return 0 below to tell
			 * download_client to retry
			 */`

Reply:
I think it's fine to not issue the event if the `fd` is `-1`, provided the only time when that happens if when `_disconnect()` is called by the user.